apps/web/scripts/kernel/handlers/shared.ts

```diff
@@ -0,0 +1,30 @@
+import { type MiniBroadcast } from '@walless/core';
+import { type MessagePayload, type PopupType } from '@walless/messaging';
+
+interface RequestContent {
+	channel: MiniBroadcast;
+	popup?: chrome.windows.Window;
+	payload: MessagePayload;
+	resolve: boolean;
+}
+
+export const requestMap: Record<string, RequestContent> = {};
```

Comment:
I think we need to make it universal, like request pool, and when we want to response instead of using `channel.postMessge` and include `requestId` inside payload, we will use another method like `messaging.response({ to: requestId, payload})`.

Reply:
I think that `requestMap` currently act as the request pool.

Reply:
we do have more than one place using this pattern, I'm not sure unified them benefits us enough, more than the effort it takes.. but I do love to see alternative implementations.